rclcpp/src/rclcpp/executor.cpp

```diff
@@ -190,14 +191,13 @@ Executor::add_callback_groups_from_nodes_associated_to_executor()
   for (auto & weak_node : weak_nodes_) {
     auto node = weak_node.lock();
     if (node) {
-      auto group_ptrs = node->get_callback_groups();
-      std::for_each(
-        group_ptrs.begin(), group_ptrs.end(),
-        [this, node](rclcpp::CallbackGroup::WeakPtr group_ptr)
+      auto node_base_type = std::dynamic_pointer_cast<rclcpp::node_interfaces::NodeBase>(node);
```

Comment:
This is a bit of bikeshedding, but I feel like `node_base` is a better name for this variable.  This is the instantiation of a type, not the type itself.  Same goes a few times below.